### PR TITLE
[arcgateway] release preview version 2024-07-31

### PR DESCRIPTION
**Pull request by yaotongms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az arcgateway: Manage gateway connection on Arc machine
az arcgateway settings: Manage Settings for Azure Arc-Enabled Server

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_Setting_scenario.py
Comment (kairu-ms) [inline src/connectedmachine/azext_connectedmachine/tests/latest/test_Setting_scenario.py:1]: Please don't change the file in other extensions.

File: src/arcgateway/azext_arcgateway/commands.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+# from azure.cli.core.commands import CliCommandType
+
+
+def load_command_table(self, _):  # pylint: disable=unused-argument
+    self.command_table["arcgateway settings update"] = Update(loader=self)
```
Comment (AllyW) [inline src/arcgateway/azext_arcgateway/commands.py:15]: import `Update`?

Comment (yaotongms) [reply src/arcgateway/azext_arcgateway/commands.py:15]: I was trying to customize 'az arcgateway settings update'. Do you know how to overwrite the command here?

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_Setting_scenario.py
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/tests/latest/test_Setting_scenario.py:1]: recovered file

File: src/arcgateway/azext_arcgateway/commands.py (hunk 1 above)
Comment (yaotongms) [reply src/arcgateway/azext_arcgateway/commands.py]: updated custom and commands files to correctly overwrite the update command.

**Review by kairu-ms — APPROVED** [review]

**Comment by yaotongms** [comment]
https://github.com/Azure/aaz/pull/606

**Pull request merged**